DAWdleUnitTesting/UnitTesting.cpp

```diff
@@ -811,4 +811,64 @@ namespace MemoryTests {
 		char buffer2[10] = "hello";
 		EXPECT_EQ(memcmp(buffer1, buffer2, 3), 0);
 	}
-}
\ No newline at end of file
+}
+
+namespace ByteBuffer {
+
+	const StrA str = "dawdleDotcom"sa;
+
+	TEST(ByteBuff, Wrap) {
+		ByteBuf test;
+		U8 testValue = 'a';
+		U8* testdata = &testValue;
+		U32 dataSize = sizeof(testdata);
+		test.wrap(testdata, dataSize);
+		EXPECT_EQ(test.bytes, 'a');
+	}
+
+	TEST(ByteBuff, has_data_left) {
+		ByteBuf test;
+		U8 testValue = 'a';
+		U8* testdata = &testValue;
+		U32 dataSize = sizeof(testdata);
+		test.wrap(testdata, dataSize);
+		EXPECT_TRUE(test.has_data_left(dataSize));
+	}
+
+	TEST(ByteBuff, skip) {
+		ByteBuf test;
+		U8 testValue = 'a';
+		U8* testdata = &testValue;
+		U32 dataSize = sizeof(testdata);
+		test.wrap(testdata, dataSize);
+		test.skip(1);
+		EXPECT_EQ(test.bytes, 'a');
+	}
+
+	TEST(ByteBuff, read_bytes) {
+		ByteBuf test;
+		U8 testValue = 'a';
+		U8* testdata = &testValue;
+		U32 dataSize = sizeof(testdata);
+		test.read_bytes(testdata, dataSize);
+		EXPECT_EQ(test.bytes, 'a');
+	}
+
+	TEST(ByteBuff, write_bytes) {
+		ByteBuf test;
+		ByteBuf result;
```

Comment:
The reason write_bytes returns a ByteBuf& is so that you can chain function calls together. For example, 
`buf.write_f32(...).write_bytes(...).write_bytes(...).write_stra("myString"sa);`
I don't think it's particularly useful to assign the result to another variable, since all that does is make an extra copy.